app/javascript/gobierto_data/webapp/components/sets/VisualizationsItem.vue

```diff
@@ -0,0 +1,223 @@
+      type: Boolean,
+      default: false
+    }
+  },
+  data() {
+    return {
+      labelVisName: I18n.t('gobierto_data.projects.visName') || "",
+      labelEdit: I18n.t('gobierto_data.projects.edit') || "",
+      labelSaveViz: I18n.t('gobierto_data.projects.saveViz') || "",
+      labelVisualize: I18n.t('gobierto_data.projects.visualize') || "",
+      labelDashboard: I18n.t('gobierto_data.projects.dashboards') || "",
+      labelSavedVisualization: I18n.t("gobierto_data.projects.savedVisualization") || "",
+      labelModifiedVizualition: I18n.t("gobierto_data.projects.modifiedVisualization") || "",
+      items: null,
+      config: {},
+      vizID: null,
+      user: null,
+      queryViz: '',
+      isVizElementSavingVisible: false,
+      name: '',
+      isQuerySavingPromptVisible: false,
+      tabs
+    }
+  },
+  watch: {
+    publicVisualizations(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.getDataVisualization(newValue);
+      }
+    },
+    privateVisualization(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.getDataVisualization(newValue);
+      }
+    },
+    vizInputFocus(newValue) {
+      if (newValue) {
+        this.$nextTick(() => this.$refs.savingDialogVizElement.inputFocus())
+      }
+    }
+  },
+  async created() {
+    const userId = getUserId()
+    if (userId) {
+      await this.getDataVisualization(this.privateVisualizations);
+      await this.getDataVisualization(this.publicVisualizations);
+    } else {
+      await this.getDataVisualization(this.publicVisualizations);
+    }
+  },
+  beforeDestroy() {
+    //Hide the string, and the buttons return to their initial state.
+    this.$root.$emit("isVizModified", false);
+    this.$root.$emit('disabledSavedVizString')
+  },
+  methods: {
+    onSaveEventHandler(opts) {
+      //Add visualization ID to opts object, we need it to update a viz saved
+      opts.vizID = Number(this.vizID)
+      opts.user = Number(this.user)
+      opts.queryViz = Number(this.queryViz)
+      // get children configuration
+      const config = this.$refs.viewer.getConfig()
+
+      this.$root.$emit("storeCurrentVisualization", config, opts);
+    },
+    updateVizName(value) {
+      const {
+        name: vizName
+      } = value;
+      this.labelValue = vizName
+      this.$root.$emit('updateVizName')
+    },
+    showChart() {
+      const showPerspective = "flex"
+      this.$root.$emit('disabledSavedVizString')
+      this.$refs.viewer.enableDisabledPerspective(showPerspective);
+    },
+    showSavingDialog() {
+      const userId = getUserId()
+      this.showVisualize = false
+      this.showResetViz = true
+      //Enable saved button
+      this.$root.$emit('enableSavedVizButton', true)
+      this.$root.$emit("isVizModified", true);
+      this.$nextTick(() => this.$refs.savingDialogVizElement.inputFocus())
+
+      if (userId !== this.user) {
+        this.isQuerySavingPromptVisible = true
+      }
```

Comment:
Has `VisualizationItem` the decision the handle the user stuff, or perhaps their parents?